Geometry/GEMGeometryBuilder/src/GEMGeometryParsFromDD.cc

```diff
@@ -58,7 +58,32 @@ void GEMGeometryParsFromDD::buildGeometry(DDFilteredView& fv,
   MuonGeometryNumbering muonDDDNumbering(muonConstants);
   GEMNumberingScheme gemNumbering(muonConstants);
 
-  bool doSuper = fv.firstChild();
+  // Check for the demonstrator geometry (only 1 chamber of GE2/1)
+  DDFilteredView fvGE2{fv};
+  int nGE21 = 0;
+  bool doSuper = fvGE2.firstChild();
+  while (doSuper) {
+    // getting chamber id from eta partitions
+    fvGE2.firstChild();
+    fvGE2.firstChild();
+    int rawidCh = gemNumbering.baseNumberToUnitNumber(muonDDDNumbering.geoHistoryToBaseNumber(fvGE2.geoHistory()));
+    GEMDetId detIdCh = GEMDetId(rawidCh);
+    if (detIdCh.station() == 2)
+      nGE21++;
+
+    // back to chambers
+    fvGE2.parent();
+    fvGE2.parent();
+    doSuper = fvGE2.nextSibling();
+  }
+  bool demonstratorGeometry = nGE21 == 1;
+
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("Geometry") << "Found " << nGE21 << " GE2/1 chambers. Demonstrator geometry on? "
+                               << demonstratorGeometry;
+#endif
+
+  doSuper = fv.firstChild();
```

Comment:
I'm not sure I understand the logic here. doSuper is now used first on GE2/1, but then is it overwritten by the original line?

Reply:
doSuper is true while there are more superchambers to process, I run through the whole sim geometry initially to see if we are in the demonstrator condition with the fvGE2 view, then start over again here with the fv to actually build the reco geometry. The doSuper variable is just being reused for the same purpose